components/portal/messages/filters.js

```diff
@@ -67,6 +67,40 @@ define(['angular'], function(angular) {
           }
         });
       };
+    })
+     .filter('filterMessagesWithInvalidDates', function() {
+      var retVal = [];
+      return function(messages) {
+          var thePresent = Date.now();
+          angular.forEach(messages, function(message) {
+            if (message.expireDate || message.goLiveDate) {
+              var addToRetVal = true;
+              if (message.goLiveDate) {
+                var goDate = new Date(message.goLiveDate);
+                var goTime = goDate.getTime();
+                if (thePresent < goTime) {
```

Comment:
:art: prefer [`moment.isBefore`](https://momentjs.com/docs/#/query/is-before/) over `Date` -> `Number` conversion, followed by compare.